components/tidb_query/src/rpn_expr/impl_math.rs

```diff
@@ -112,4 +180,85 @@ mod tests {
             assert_eq!(output, expect_output, "{:?}", arg);
         }
     }
+
+    #[test]
+    fn test_floor_real() {
+        let cases = vec![
+            (3.0, 3.5),
```

Comment:
Hm, do we need to worry about floating point precision here?

For example:

```rust
fn main() {
    let maybe = 16140901064495871255;
    assert_eq!(maybe, (maybe as f64) as u64);
}
```

Yields:

```
   Compiling playground v0.0.1 (/playground)
    Finished dev [unoptimized + debuginfo] target(s) in 0.65s
     Running `target/debug/playground`
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `16140901064495871255`,
 right: `16140901064495872000`', src/main.rs:3:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

Reply:
`floor` returns the largest integer less than or equal to a number (an `f64`). Its behavior is as same as https://doc.rust-lang.org/std/primitive.f64.html#method.floor. So we could ignore floating-point precision here.

Reply:
Hm, ok. I did some investigation and it seems like REAL in SQL is also intended to be a float. :) Thanks for looking at this.